src/lm.jl

```diff
@@ -63,18 +63,18 @@ end
 
 cholfact(x::LinearModel) = cholfact(x.pp)
 
-function fit{LmRespT<:LmResp,LinPredT<:LinPred}(::Type{LinearModel{LmRespT,LinPredT}},
-    X::AbstractMatrix, y::Vector)
+function fit{LmRespT<:LmResp,LinPredT<:LinPred, T<:FP}(::Type{LinearModel{LmRespT,LinPredT}},
+    X::@compat(Union{Matrix{T},SparseMatrixCSC{T}}), y::FPVector)
     rr = LmRespT(y)
-    pp = LinPredT(X)
+    pp = LinPredT(full(X))
     installbeta!(delbeta!(pp, rr.y))
     updatemu!(rr, linpred(pp, 0.0))
     LinearModel(rr, pp)
 end
-function fit(::Type{LinearModel}, X::Matrix, y::Vector)
-    T = promote_type(eltype(X), eltype(y), Float32)
```

Comment:
@andreasnoack You wrote these lines in https://github.com/pkofod/GLM.jl/commit/1ef71066b5c60e3419d632e489b00ca1c783a8a6. Why are they different from the corresponding code for GLMs? Should we keep the logic, even if it was broken (`yy` wasn't used)?